### PR TITLE
Implement sorting for resource tiles

### PR DESCRIPTION
**Pull request by JammingBen**
image

## Related Issue

- Fixes https://github.com/owncloud/web/issues/8370

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

File: packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts [hunk 1]
```diff
@@ -76,6 +69,18 @@ export const useResourcesViewDefaults = <T, TT, TU extends any[]>(
   const currentTilesSize = computed((): string => String(currentTilesSizeQuery.value))
   const viewSize = useViewSize(currentTilesSize)
 
+  const fields = computed((): SortField[] => {
+    if (unref(viewMode) === ViewModeConstants.tilesView.name) {
+      return determineResourceTilesSortFields(unref(storeItems)[0])
+    }
+    return determineResourceTableSortFields(unref(storeItems)[0])
+  })
```
Comment (dschmidt) [inline packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts:77]: eeeks ... why is this needed now?

File: packages/web-pkg/tests/unit/composables/router/useRouteQuery.spec.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ describe('useRouteQuery', () => {
 
     // FIXME: why do we have to wait for so many ticks?
     // Why don't we have to do that for any other expectation in the whole file?!
-    for (let i = 0; i < 32; i++) {
+    for (let i = 0; i < 33; i++) {
```
Comment (dschmidt) [inline packages/web-pkg/tests/unit/composables/router/useRouteQuery.spec.ts:44]: I hate everything about this (and I know I've introduced this code)

File: packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts (hunk 1 above)
Comment (JammingBen) [reply packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts:77]: Because the Tile view has different sorting fields then the table. Although I'm still debating if we keep one `determineSortFields` and let that handle the fields.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oC10</strong> https://drone.owncloud.com/owncloud/web/32678/73/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingNotifications-shareWithUsers_feature-L40.png
  


</p></details>
:boom: The oC10IntegrationNotifications tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/32727/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/participant-management-alice-2023-2-13-01-33-12.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/participant-management-anonymous-2023-2-13-01-33-47.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/participant-management-brian-2023-2-13-01-33-22.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/participant-management-carol-2023-2-13-01-33-35.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/participant-management-marie-2023-2-13-01-33-26.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/participant-management-richard-2023-2-13-01-33-29.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32727/tracing/unstructured-collection-of-testable-space-interactions-alice-2023-2-13-01-34-34.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

**Review by fschade — APPROVED** [review]
🎸 as usual, works as expected, just two minor nitpicks.

File: packages/web-app-files/src/components/FilesList/ResourceTiles.vue [hunk 3]
```diff
@@ -1,5 +1,39 @@
 <template>
   <div>
+    <div v-if="!!sortFields.length" class="oc-tile-sorting oc-border-b oc-mb-m oc-pb-s">
```
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-app-files/src/components/FilesList/ResourceTiles.vue]: I just got used to doing it like this 😅 If this would be the return of a method, it would be technically correct. But yeah, here, it is superfluous.

**Review by kulmann — CHANGES_REQUESTED** [review]
changing the sorting multiple times in a row somehow starts to switch to sorting option that I didn't select. 🙈  example (left: clicked option, right: active sorting after click)
```
A-Z preselected
Z-A -> Z-A
Newest -> Newest
Oldest -> Oldest
**Z-A -> Newest**
```

**Comment by kulmann** [comment]
> > changing the sorting multiple times in a row somehow starts to switch to sorting option that I didn't select. 🙈 example (left: clicked option, right: active sorting after click)
> > ```
> > A-Z preselected
> > Z-A -> Z-A
> > Newest -> Newest
> > Oldest -> Oldest
> > **Z-A -> Newest**
> > ```
> 
> Yes, see [#8410 (comment)](https://github.com/owncloud/web/pull/8410#issuecomment-1428011599) 😄

Ah. Lol. Well 💩 😆 But fine then 👍

File: packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts [hunk 4]
```diff
@@ -76,6 +69,14 @@ export const useResourcesViewDefaults = <T, TT, TU extends any[]>(
   const currentTilesSize = computed((): string => String(currentTilesSizeQuery.value))
   const viewSize = useViewSize(currentTilesSize)
 
+  const fields = computed((): SortField[] => {
```
Comment (kulmann) [inline packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts:72]: Why not call it `sortFields`? 😁

Comment (kulmann) [reply packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts:72]: in the context of `useSort` the name `fields` is clear, but outside of that composable it's not IMO.

Comment (JammingBen) [reply packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts:72]: Because it was named `fields` before 🙈  I need to check where we use this, maybe I can easily rename it without changing it all over the place.

File: packages/web-app-files/src/views/spaces/Projects.vue [hunk 5]
```diff
@@ -96,12 +104,21 @@ export default defineComponent({
...
-    const spaces = computed(
-      () =>
-        store.getters['runtime/spaces/spaces']
-          .filter((s) => isProjectSpaceResource(s))
-          .sort((a, b) => a.name.localeCompare(b.name)) || []
+    const runtimeSpaces = computed(
+      () => store.getters['runtime/spaces/spaces'].filter((s) => isProjectSpaceResource(s)) || []
     )
+
+    const sortFields = [availableSortFields[0], availableSortFields[1]]
```
Comment (kulmann) [inline packages/web-app-files/src/views/spaces/Projects.vue:111]: Would love to see the other sort options for project spaces as well... for size we could utilize the used quota. And the `mdate` actually has a value present... not sure how it's updated though.

Comment (kulmann) [reply packages/web-app-files/src/views/spaces/Projects.vue:111]: (fine as a followup if you want to, but in that case please create an issue)

Comment (kulmann) [reply packages/web-app-files/src/views/spaces/Projects.vue:111]: the `size` prop for `SpaceResource`s is empty... so you could just write the `spaceQuota?.used` into the `size` prop in `buildSpaceResource`. That should make the size work.

Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/Projects.vue:111]: I've created https://github.com/owncloud/web/issues/8439. `mdate` would also be nice, but needs some adjustments as well.

File: packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts (hunk 4 above)
Comment (JammingBen) [reply packages/web-app-files/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts]: Seems that only a few locations were affected :)

**Review by kulmann — APPROVED** [review]
🥳
